app/uk/gov/hmrc/tariffclassificationfrontend/views/report_sla.scala.html

```diff
@@ -0,0 +1,137 @@
+@*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+@import uk.gov.hmrc.tariffclassificationfrontend.controllers.routes
+@import uk.gov.hmrc.tariffclassificationfrontend.models.request.AuthenticatedRequest
+@import uk.gov.hmrc.tariffclassificationfrontend.models.{InstantRange, Queue, ReportResult}
+@import uk.gov.hmrc.tariffclassificationfrontend.utils.Dates
+@import uk.gov.hmrc.tariffclassificationfrontend.views.html.includes.main
+@import uk.gov.hmrc.tariffclassificationfrontend.views.{Histogram, HistogramBucket, HistogramBucketInterval, Report}
+@(filter: InstantRange, results: Seq[ReportResult], queues: Seq[Queue])(implicit request: AuthenticatedRequest[_], messages: Messages, appConfig: AppConfig)
+
+@main(bodyClasses = None) {
+    <a id="back-link" class="link-back mt-0" href="@routes.ReportingController.getReportCriteria(Report.SLA.toString)">
+        Back to reports
+    </a>
+
+    <h1 class="heading-large" id="report_sla-heading">SLA Report</h1>
+
+    <dl class="dl-mb-2 mt-3 mb-1">
+        <div class="grid-row">
+            <dt class="column-one-quarter">
+            From
+            </dt>
+            <dd id="report_sla-from_date" class="column-three-quarters">
+            @Dates.format(filter.min)
+            </dd>
+        </div>
+
+        <div class="grid-row">
+            <dt class="column-one-quarter">
+                To
+            </dt>
+            <dd id="report_sla-to_date" class="column-three-quarters">
+                @Dates.format(filter.max)
+            </dd>
+        </div>
+    </dl>
+
+    @defining(Seq(
```

Comment:
I was expecting to see the processing of the report data (histograms) pulled out into some sort of SLAReport object with the UI just responsible for rendering it.  Won't we have to repeat this logic for the 'csv' version of the report?

Reply:
Yep, we can pull any duplicates out as part of that. I suspect we can just wrap the histogram in a class with some useful methods specific to the report

Reply:
@SteveSugden I've pulled this out into an SLAReport class. Should make the CSV bit a lot easier